drivers/btrfs/btrfs.go

```diff
@@ -627,7 +627,12 @@ func (d *Driver) Remove(id string) error {
 	d.updateQuotaStatus()
 
 	if err := subvolDelete(d.subvolumesDir(), id, d.quotaEnabled); err != nil {
-		return err
+		if d.quotaEnabled {
+			return err
+		}
+		// If quota is not enabled, fallback to rmdir syscall to delete subvolumes.
+		// This would allow unprivileged user to delete their owned subvolumes
+		// in kernel >= 4.18 without user_subvol_rm_alowed mount option.
```

Comment:
Is this going to be OK if the kernel is less than 4.18?

Reply:
I would figure that container/storage with btrfs does not work well on this old of kernel anyways.

Reply:
prior to 4.18, system.EnsureRmoveall() would return error since subvolume cannot be removed (though subvolume's contents are cleaned up). So the difference is from where error retuens and I don't think this would cause any problem.